src/java.base/share/classes/java/lang/reflect/code/writer/OpWriter.java

```diff
@@ -54,10 +54,19 @@ static final class GlobalValueBlockNaming implements Function<CodeItem, String>
             this.gn = new HashMap<>();
         }
 
+        private String blockId(Block b) {
+            if (b.index() < 0) return "__" + blockOrdinal++;
```

Comment:
I assumed such case is possible based on the implementation comment:
```
    // Block is inoperable when < 0 i.e., when partially built
    int index = -1;
```
However it can simplify if OpWriter could never be called on a partially built block.

Reply:
It's not possible to traverse a partially built code model via the builders, since access is limited, and a builder can never produce partial or structurally incorrect models, since it would fail otherwise.